frontend/src/app.js

```diff
@@ -33,79 +29,71 @@ export class App extends React.Component {
-            view.widget !== "jenkins-analysis-view" && (
-            <li className="pf-v5-c-nav__item" key={view.id}>
-              <NavLink to={`/view/${view.id}`} className="pf-v5-c-nav__link">{view.title}</NavLink>
-            </li>
-            )
-          ))}
-        </NavList>
-      </Nav>
-    );
-
     return (
-      <React.Fragment>
-        <IbutsuPage eventEmitter={this.eventEmitter} navigation={navigation}>
-          <Routes>
-            <Route path="*" element={<Dashboard eventEmitter={this.eventEmitter} />} />
-            <Route path="/runs" element={<ElementWrapper routeElement={RunList} eventEmitter={this.eventEmitter} />} />
-            <Route path="/results" element={<ElementWrapper routeElement={ResultList} eventEmitter={this.eventEmitter} />} />
-            <Route path="/reports" element={<ElementWrapper routeElement={ReportBuilder} eventEmitter={this.eventEmitter} />} />
-            <Route path="/runs/:id" element={<ElementWrapper routeElement={Run} />} />
-            <Route path="/results/:id" element={<ElementWrapper routeElement={Result} />} />
-            <Route path="/view/:id" element={<ElementWrapper routeElement={View} />} />
-          </Routes>
-        </IbutsuPage>
-      </React.Fragment>
+      <Routes>
+        <Route
+          path=""
+          element={<ElementWrapper routeElement={IbutsuPage} eventEmitter={this.eventEmitter} />}
+        />
+        <Route
+          path=":project_id/*"
+          element={<ElementWrapper routeElement={IbutsuPage} eventEmitter={this.eventEmitter} />}
+        >
+
+          {/* Nested project routes */}
+          <Route
+            path="dashboard/:dashboard_id"
+            element={
+              <ElementWrapper routeElement={Dashboard} eventEmitter={this.eventEmitter} />
+            }
+          />
+          <Route
+            path="dashboard/*"
+            element={<ElementWrapper routeElement={Dashboard} eventEmitter={this.eventEmitter} />}
+          />
+
+
+          <Route
+            path="runs"
+            element={
+              <ElementWrapper routeElement={RunList} eventEmitter={this.eventEmitter} />
+            }
+          />
+          <Route
+            path="runs/:run_id"
+            element={<ElementWrapper routeElement={Run} />}
+          />
+
+          <Route
+            path="results"
+            element={<ElementWrapper routeElement={ResultList} eventEmitter={this.eventEmitter} />}
+          />
+          <Route
+            path="results/:result_id"
+            element={<ElementWrapper routeElement={Result} />}
+          />
+
+          <Route
+            path="reports"
+            element={<ElementWrapper routeElement={ReportBuilder} eventEmitter={this.eventEmitter} />}
+          />
+
+          <Route
+            path="view/:view_id"
+            element={<ElementWrapper routeElement={View} />}
+          />
+
+          {/* <Route path="*" element={<Navigate to="" replace />} /> */}
```

Comment:
I think I want to uncomment this to provide a diaper for bad URLs to redirect. There's one in base.js to redirect to `/project`, I had issues with this one while getting all the nested routing setup correctly.

I'll uncomment it and test with it now, in this case it will also go to `/project` but because of thee routing I think the `to` prop will be empty

Reply:
Played with this but couldn't get it to behave like I wanted anyway.

I added catch statements to the HTTP requests to at least stop runtime errors if a bad UUID is entered in the URL.